lib/dsl/validator.js

```diff
@@ -33,6 +33,7 @@ const ALPHABETIC = /^[A-Za-z]+$/;
 const ALPHABETIC_LOWER = /^[a-z]+$/;
 const ALPHANUMERIC = /^[A-Za-z][A-Za-z0-9]*$/;
 const ALPHANUMERIC_DASH = /^[A-Za-z][A-Za-z0-9-]*$/;
+const LANGUAGE_PATTERN = /^[a-z]+(-[A-Za-z0-9]+)*$/;
```

Comment:
Should be `/^[a-z]+(-[A-Za-z0-9]+)?$/`, because we don't want something like `ab-cd-ef`

Reply:
I used to create a more complex regex to match more cases defined in rfc5646.
//https://tools.ietf.org/html/rfc5646#section-4.4.1
const LANGUAGE_PATTERN = /^[a-z]{2,8}(-[A-Za-z0-9]{1,9}){0,8}$/;

Which should be allowed to have languages ab-cd-ef, ref:
https://www.w3.org/International/articles/bcp47/

Reply:
I didn't know that, you're right! Thanks for pointing it out.